ethclient/ethclient.go

```diff
@@ -0,0 +1,586 @@
+// Copyright 2016 The go-ethereum Authors
```

Comment:
Honestly, I'm not sure exactly what this file is doing. But it seems to mainly be a drop in from geth.

Reply:
Yup this is just a drop in from geth. Since there were a few small changes to the block format in coreth, using geth's ethclient will not handle the extra fields, so we need to have our own version both to correctly parse the extra data and also in order to make sure that the client will produce the correct hashes of the blocks that it does retrieve. Using this version of ethclient fixes #46